pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -529,6 +529,16 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 	}
 
 	if cd.Spec.Installed {
+		// set installedTimestamp for adopted clusters
+		if cd.Status.InstalledTimestamp == nil {
+			now := metav1.Now()
+			cd.Status.InstalledTimestamp = &now
+			if err := r.Status().Update(context.TODO(), cd); err != nil {
+				cdLog.WithError(err).Log(controllerutils.LogLevel(err), "could not set cluster installed timestamp")
+				return reconcile.Result{Requeue: true}, nil
+			}
+		}
```

Comment:
I'm worried this leads to inaccurate data, could we get into any trouble reporting install times that can't match reality? We really don't know when an adopted cluster was installed. 

What do you think about just NOT reporting the metric for these?

Or if installed=true but no installedTimestamp, use the clusterdeployment.CreationTimestamp.